.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,899 @@
+        . setenv.sh
+        mwc.pl -type gnuace $TAO_ROOT/tests/Hello
+    - name: build TAO Hello test
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        make -j2 dir=$TAO_ROOT/tests/Hello
+    - name: set up Modeling tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        cd tools/modeling/tests
+        ./setup.pl
+        cd -
+        mwc.pl -type gnuace -features built_in_topics=0 tools/modeling/tests/modeling_tests.mwc
+    - name: build Modeling tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        make -j2 -C tools/modeling/tests
+    - name: run OpenDDS tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        mkdir workspace
+        cd workspace
+        cat > config.xml <<EOF
+        <autobuild>
+        <configuration>
+        <variable name="log_file" value="ubuntu-18_04_java_no-bit_autobuild_output.log"/>
+        <variable name="log_root" value="$DDS_ROOT/workspace"/>
+        <variable name="project_root" value="$DDS_ROOT"/>
+        <variable name="root" value="$DDS_ROOT/workspace"/>
+        <variable name="junit_xml_output" value="Tests"/>
+        </configuration>
+        <command name="log" options="on"/>
+        <command name="git" options="dir=$GITHUB_WORKSPACE/OpenDDS log -1"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_make_version"/>
+        <command name="check_compiler" options="gcc"/>
+        <command name="print_autobuild_config"/>
+        <command name="auto_run_tests" options="dir=$GITHUB_WORKSPACE/OpenDDS -Config RTPS -Config CXX11 -Config RAPIDJSON -Config NO_BUILT_IN_TOPICS -a -l java/tests/dcps_java_tests.lst -l tools/modeling/tests/modeling_tests.lst"/>
+        <command name="log" options="off"/>
+        <command name="process_logs" options="prettify"/>
+        </autobuild>
+        EOF
+        $GITHUB_WORKSPACE/autobuild/autobuild.pl "$GITHUB_WORKSPACE/OpenDDS/workspace/config.xml"
+    - name: upload Full log output
+      uses: actions/upload-artifact@v2
+      with:
+        name: ubuntu-18_04_java_no-bit_autobuild_output.log_Full.html
+        path: OpenDDS/workspace/ubuntu-18_04_java_no-bit_autobuild_output.log_Full.html
+    - name: check results
+      shell: bash
+      run: |
+        cat "$GITHUB_WORKSPACE/OpenDDS/workspace/latest.txt"
+        grep -q 'Failures: 0' "$GITHUB_WORKSPACE/OpenDDS/workspace/latest.txt"
+    - name: Publish Unit Test Results
+      uses: mikepenz/action-junit-report@v1
+      if: always()
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        check_name: "Test Results: ubuntu-18_04_java_no-bit"
+        report_paths: OpenDDS/workspace/Tests_JUnit.xml
+
+  # Windows
+
+  windows-2019_defaults_artifact:
+
+    runs-on: windows-2019
+
+    steps:
+    - name: install openssl-windows & xerces-c
```

Comment:
Would recommend to use run-vcpkg, that uses caching, as far as I know the default vcpkg doesn't do that, see https://github.com/DOCGroup/ACE_TAO/blob/a72619e2ffcb37bc821c76b1e30b6886c2664003/.github/workflows/windows.yml#L99

Reply:
I've tried it, and it looks like it had a cache miss for the case when the cache key was identical, so it really didn't change the overall time by much. Perhaps I just need to run the workflow again with the same cache key, since it looks like updating the cache is sometimes post-build step?

I am, however, glad that I read the documentation for actions/cache, which is probably what we want to be using rather than artifacts to copy ACE/TAO files between jobs in the workflow. This is especially true if it will let you pull cached samples between workflow runs and lets you pull the cached values from master/main, which is how I understand the documentation. So I'll be looking into using that to avoid having to build the artifacts at all as part of normal pull request runs, which was part of the motivation for setting it up this way.

Reply:
First time run-vcpkg it makes the cache, in next runs it saves time, see the ACE-TAO actions